synapse/config/registration.py

```diff
@@ -15,19 +15,43 @@
 
 from ._base import Config
 
+from synapse.util.stringutils import random_string_with_symbols
+
+import distutils.util
+
 
 class RegistrationConfig(Config):
 
     def __init__(self, args):
         super(RegistrationConfig, self).__init__(args)
-        self.disable_registration = args.disable_registration
+
+        self.disable_registration = bool(
+            distutils.util.strtobool(str(args.disable_registration))
```

Comment:
It can either be the a bool or string depending on exactly where it was set. This is mainly to make `--disable-registration=false` work, which sets `args.disable_registration` to `"false"`.

I'll add a comment to make this clearer.